v-next/hardhat-ignition-viem/src/types.ts

```diff
@@ -0,0 +1,91 @@
+import type {
+  IgnitionModuleResult,
+  StrategyConfig,
+  IgnitionModule,
+  DeploymentParameters,
+  DeployConfig,
+  ContractAtFuture,
+  ContractDeploymentFuture,
+  ContractFuture,
+} from "@ignored/hardhat-vnext-ignition-core";
+import type {
+  ContractAbis,
+  GetContractReturnType,
+} from "@ignored/hardhat-vnext-viem/types";
+
+export interface ViemIgnitionHelper {
+  type: "viem";
+
+  deploy<
+    ModuleIdT extends string,
+    ContractNameT extends string,
+    IgnitionModuleResultsT extends IgnitionModuleResult<ContractNameT>,
+    StrategyT extends keyof StrategyConfig = "basic",
+  >(
+    ignitionModule: IgnitionModule<
+      ModuleIdT,
+      ContractNameT,
+      IgnitionModuleResultsT
+    >,
+    options?: {
+      parameters?: DeploymentParameters | string;
+      config?: Partial<DeployConfig>;
+      defaultSender?: string;
+      strategy?: StrategyT;
+      strategyConfig?: StrategyConfig[StrategyT];
+      deploymentId?: string;
+      displayUi?: boolean;
+    },
+  ): Promise<
+    IgnitionModuleResultsToViemContracts<ContractNameT, IgnitionModuleResultsT>
+  >;
+}
+
+export type IgnitionModuleResultsToViemContracts<
+  ContractNameT extends string,
+  IgnitionModuleResultsT extends IgnitionModuleResult<ContractNameT>,
+> = {
+  [resultKey in keyof IgnitionModuleResultsT]: ToContractType<
+    IgnitionModuleResultsT,
+    resultKey
+  >;
+};
+
+type ToContractType<
+  IgnitionModuleResultsT extends IgnitionModuleResult<string>,
+  ResultKey extends keyof IgnitionModuleResultsT,
+> = IgnitionModuleResultsT[ResultKey] extends
+  | ContractDeploymentFuture
+  | ContractAtFuture
+  ? GetContractReturnType<AbiOf<IgnitionModuleResultsT[ResultKey]>>
+  : LookupContractName<
+        IgnitionModuleResultsT,
+        ResultKey
+      > extends keyof ContractAbis
```

Comment:
The switching of the type to `ContractAbis` is the only change in these `ToContractType` types.